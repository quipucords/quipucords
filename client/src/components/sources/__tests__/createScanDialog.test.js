import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import CreateScanDialog from '../createScanDialog';

describe('CreateScanDialog Component', () => {
  const generateEmptyStore = () => configureMockStore()({ scans: {} });

  it('should render a basic component', () => {
    const store = generateEmptyStore();
    const props = {
      show: true,
      sources: [{ name: 'test name' }]
    };

    const component = mount(<CreateScanDialog {...props} />, { context: { store } });

    expect(component.render()).toMatchSnapshot();
  });
});
