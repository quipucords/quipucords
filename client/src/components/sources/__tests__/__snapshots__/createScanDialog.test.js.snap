// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateScanDialog Component should render a basic component 1`] = `
<div
  role="dialog"
>
  <div
    class="fade modal-backdrop in"
  />
  <div
    class="fade in modal"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
        role="document"
      >
        <div
          class="modal-header"
        >
          <button
            aria-hidden="true"
            aria-label="Close"
            class="close"
            type="button"
          >
            <span
              aria-hidden="true"
              class="pficon pficon-close"
            />
          </button>
          <h4
            class="modal-title"
          >
            Scan
          </h4>
        </div>
        <div
          class="modal-body"
        />
        <div
          class="container-fluid"
        >
          <form
            class="form-horizontal"
          >
            <div
              class="form-group"
            >
              <label
                class="col-sm-3 control-label"
                for="scanName"
              >
                Name
              </label>
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control"
                  id="scanName"
                  name="scanName"
                  placeholder="Enter a name for the scan"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="col-sm-3 control-label"
              >
                Sources
              </label>
              <div
                class="col-sm-9"
              >
                <textarea
                  class="quipucords-form-control form-control"
                  readonly=""
                  rows="1"
                  type="textarea"
                >
                  test name
                </textarea>
              </div>
            </div>
          </form>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn-cancel btn btn-default"
            type="button"
          >
            Cancel
          </button>
          <button
            class="btn btn-primary"
            disabled=""
            type="submit"
          >
            Scan
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
